shim/src/handle_vm_kvm_check_extension.c

```diff
@@ -24,17 +24,75 @@
  * SOFTWARE.
  */
 
+#include <debug.h>
+#include <g_mut_hndl.h>
+#include <kvm_constants.h>
+#include <mv_constants.h>
 #include <mv_types.h>
+#include <platform.h>
 
 /**
  * <!-- description -->
  *   @brief Handles the execution of kvm_check_extension.
  *
  * <!-- inputs/outputs -->
+ *   @param mut_userargs as a input from the user
+ *   @param pmut_ret as a output to the user to check the extension is supported or not
  *   @return SHIM_SUCCESS on success, SHIM_FAILURE on failure.
  */
 NODISCARD int64_t
-handle_vm_kvm_check_extension(void) NOEXCEPT
+handle_vm_kvm_check_extension(unsigned long mut_userargs, uint32_t *const pmut_ret) NOEXCEPT
 {
+    platform_expects(MV_INVALID_HANDLE != g_mut_hndl);
+    platform_expects(NULL != pmut_ret);
+
+    switch (mut_userargs) {
+        case KVM_CAP_EXT_CPUID: {
+            *pmut_ret = (uint32_t)1;
+        } break;
```

Comment:
The `break;` should be inside the curly braces:
```
        case KVM_CAP_EXT_CPUID: {
            *pmut_ret = (uint32_t)1;
            break;
        }
```
Have you tried to use `touch();` instead for the fallthrough cases?
```
        ...
        case KVM_CAP_JOIN_MEMORY_REGIONS_WORKS: {
            touch();
        }
        case KVM_CAP_IMMEDIATE_EXIT: {
            *pmut_ret = (uint32_t)1;
            break;
        }
```

Reply:
Getting compilation errors..when use the above method to implement

Reply:
/home/developer/working/microv/shim/src/handle_vm_kvm_check_extension.c:53:9: error: unannotated fall-through between switch labels [-Werror,-Wimplicit-fallthrough]
        case KVM_CAP_GET_TSC_KHZ: {
        ^
/home/developer/working/microv/shim/src/handle_vm_kvm_check_extension.c:53:9: note: insert '__attribute__((fallthrough));' to silence this warning
        case KVM_CAP_GET_TSC_KHZ: {
        ^
        __attribute__((fallthrough)); 
/home/developer/working/microv/shim/src/handle_vm_kvm_check_extension.c:53:9: note: insert 'break;' to avoid fall-through
        case KVM_CAP_GET_TSC_KHZ: {
        ^
        break; 
/home/developer/working/microv/shim/src/handle_vm_kvm_check_extension.c:56:9: error: unannotated fall-through between switch labels [-Werror,-Wimplicit-fallthrough]
        case KVM_CAP_TSC_DEADLINE_TIMER: {
        ^
/home/developer/working/microv/shim/src/handle_vm_kvm_check_extension.c:56:9: note: insert '__attribute__((fallthrough));' to silence this warning
        case KVM_CAP_TSC_DEADLINE_TIMER: {
        ^
        __attribute__((fallthrough));